### PR TITLE
[21134] Connect markdown prettifier for OMZ models description

### PR DESCRIPTION
**Pull request by yatarkan**
### All Submissions:
* [x] Create a meaningful name of the PR with reference to JIRA. Example: *[43714] [NIGHTLY] Take correct tag for snyk scans*
* [x] JIRA tickets:
  * 21134 - Connect markdown prettifier for Downloader models description
* [x] Set assignee to the PR.
* [x] Add corresponding labels to the PR. 
* [x] Set appropriate milestone to the PR.
* [ ] Added tests for new functionality (unit / e2e)
* [x] Followed code style guidelines:
  * Code covered with typings
  * No debug printing in production code
  * No commented blocks of code in production code

### CODE REVIEW

File: client/src/app/modules/project/components/target-list/target-list.component.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { TargetMachineItem, TargetMachineStatusNames } from '@shared/models/pipe
 
 export function getAvailableDevices(targetMachine: TargetMachineItem): string {
   const devices: string[] = map(targetMachine.devices, 'deviceName') || [];
-  return devices.join(', ');
+  return devices.toString();
```
Comment (as-suvorov) [inline client/src/app/modules/project/components/target-list/target-list.component.ts:10]: what is the motivation to change `join(', ')` to `toString()`?

File: client/src/app/shared/components/markdown-text/markdown.service.ts [hunk 2]
```diff
@@ -1,20 +1,19 @@
 import { Injectable, SecurityContext } from '@angular/core';
 import { DomSanitizer } from '@angular/platform-browser';
 
-import { IMarkdownParser, IMarkdownParserOptions } from './index';
+import { IMarkdownParser, IMarkdownParserOptions } from './markdown-parser';
```
Comment (as-suvorov) [inline client/src/app/shared/components/markdown-text/markdown.service.ts:4]: wouldn't importing this interfaces from `markdown-parser` would result in statically linking markdown libraries which were intended to be loaded lazily?

File: client/src/app/shared/components/markdown-text/markdown-parser.ts [hunk 3]
```diff
@@ -3,21 +3,32 @@ import hljs from 'highlight.js/lib/core';
...
-  return marked.parse(noYamlHeaderText, options.highlight ? highlightConfig : null);
+  return marked.parse(noYamlHeaderText, options.highlight ? highlightConfig : commonConfig);
 }
 
+const commonConfig = {
+  breaks: true,
+};
+
 const highlightConfig = {
+  ...commonConfig,
```
Comment (as-suvorov) [inline client/src/app/shared/components/markdown-text/markdown-parser.ts:31]: I would be better do not link this configs

File: client/src/app/shared/utils/array-toString.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Array.prototype.toString = function (): string {
+  return this.join(', ');
+};
```
Comment (as-suvorov) [inline client/src/app/shared/utils/array-toString.ts:3]: let's discuss this approach, according to what happened with prototype.js there are some concerns

**Review by as-suvorov — CHANGES_REQUESTED** [review]
Comments to be discussed

File: client/src/app/shared/components/markdown-text/markdown.service.ts (hunk 2 above)
Comment (yatarkan) [reply client/src/app/shared/components/markdown-text/markdown.service.ts:4]: Discussed, decided to make usual import

**Review by as-suvorov — APPROVED** [review]

File: client/src/app/shared/utils/array-toString.ts (hunk 4 above)
Comment (yatarkan) [reply client/src/app/shared/utils/array-toString.ts:3]: Agreed

File: client/src/app/shared/components/markdown-text/markdown-parser.ts (hunk 3 above)
Comment (yatarkan) [reply client/src/app/shared/components/markdown-text/markdown-parser.ts:31]: Fixed

File: client/src/app/shared/components/markdown-text/markdown.service.ts (hunk 2 above)
Comment (yatarkan) [reply client/src/app/shared/components/markdown-text/markdown.service.ts:4]: Fixed

File: client/src/app/shared/utils/array-toString.ts (hunk 4 above)
Comment (yatarkan) [reply client/src/app/shared/utils/array-toString.ts:3]: Fixed, removed

**Pull request merged**